meiga/result.py

```diff
@@ -135,11 +124,11 @@ def unwrap_or_else(
                         on_failure()
                     else:
                         on_failure(self.value)
-            return failure_value
+            return cast(TEF, failure_value)
```

Comment:
And here I presume should be either an `assert failure_value is not None`, or... another TypeVar/Union in type-hint? To indicate that if the exact `failure_value` is returned, be it `TEF` or `None`?..

Reply:
We use `cast(TEF, failure_value)` to prevent the following mypy error:

```python
error: Incompatible return value type (got "Optional[TEF]", expected "Union[TS, TEF]")
```

Sorry, but I did not understand the solution you propose, feel free to create a PR with your patch.